snapcraft/models/project.py

```diff
@@ -533,6 +534,34 @@ def _validate_platform_set(cls, values: Mapping[str, Any]) -> Mapping[str, Any]:
 
         return values
 
+    @classmethod
+    def from_architectures(
+        cls,
+        architectures: list[str | Architecture],
+    ) -> dict[str, Self]:
+        """Convert a core22 architectures configuration to core24 platforms."""
+        platforms: dict[str, Self] = {}
+        for architecture in architectures:
+            if isinstance(architecture, str):
+                build_on = build_for = [architecture]
```

Comment:
I wonder if typing `build_on` and `build_for` would let you drop the subsequent `type: ignore`s.
```suggestion
                build_on: list[str] | str = build_for: list[str] | str = [architecture]
```

Reply:
That doesn't work because mypy is complaining that it's not a `UniqueStrList`. However, I've popped in a change that uses 5 casts, but eliminates all the type checker silencing.